src/arithmetic/traits.rs

```diff
@@ -55,8 +55,8 @@ where
 }
 
 pub trait BitManipulation {
-    fn set_bit(self: &mut Self, bit: usize, bit_val: bool);
-    fn test_bit(self: &Self, bit: usize) -> bool;
+    fn set_bit(self, bit: usize, bit_val: bool);
```

Comment:
I think this is a wrong fix. Was it produced by nightly clippy? It should be `&mut self` for set_bit and `&self` for test_bit.

Reply:
Thanks for the review, and especially thanks for catching my mistake. I learned a few things from this mistake :)

Changed to borrow self as suggested.